packages/commonwealth/client/scripts/controllers/server/sessions.ts

```diff
@@ -1,24 +1,184 @@
 import { ethers } from 'ethers';
+import { Keyring } from '@polkadot/api';
+import { mnemonicGenerate } from '@polkadot/util-crypto';
+import { ChainBase } from '../../../../../common-common/src/types';
+import { DirectSecp256k1HdWallet } from '@cosmjs/proto-signing';
+import * as solw3 from '@solana/web3.js';
+import { KeyPair } from 'near-api-js';
+import { PublicKey } from 'near-api-js/lib/utils';
 
-class SessionsController {
-  wallets: Record<string, ethers.Wallet>;
-  // TODO: store expiration time on wallets[]
+
+
+abstract class ISessionController<ChainIdType extends string | number | symbol> {
+  abstract getAddress(chainId: ChainIdType): string | null;
+  abstract getOrCreateAddress(chainId: ChainIdType): Promise<string>;
+}
+
+
+class EthereumSessionController extends ISessionController<number> {
```

Comment:
Might make sense to have a generic class that gets overridden to reduce the amount of boilerplate here?

Reply:
Yep, I've removed a load of boilerplate here.